drivers/adc/adc_ads114s0x.c

```diff
@@ -0,0 +1,1025 @@
+
+	LOG_DBG("sending command 0x%02X", command);
+	int result = spi_write_dt(bus, &tx);
+
+	if (result != 0) {
+		LOG_ERR("spi_write failed with error %i", result);
+		return result;
+	}
+
+	return 0;
+}
+
+static int ads114s0x_channel_setup(const struct device *dev,
+				   const struct adc_channel_cfg *channel_cfg)
+{
+	const struct ads114s0x_config *config = dev->config;
+	uint8_t input_mux = 0;
+	uint8_t reference_control = 0;
+	uint8_t data_rate = 0;
+	uint8_t gain = 0;
+	int result;
+	enum ads114s0x_register register_addresses[4];
+	uint8_t values[ARRAY_SIZE(register_addresses)];
+
+	ADS114S0X_REGISTER_INPMUX_SET_DEFAULTS(gain);
+	ADS114S0X_REGISTER_REF_SET_DEFAULTS(reference_control);
+	ADS114S0X_REGISTER_DATARATE_SET_DEFAULTS(data_rate);
+	ADS114S0X_REGISTER_PGA_SET_DEFAULTS(gain);
+
+	if (channel_cfg->channel_id != 0) {
+		LOG_ERR("only one channel is supported");
+		return -EINVAL;
+	}
+
+	switch (channel_cfg->reference) {
+	case ADC_REF_INTERNAL:
+		/* disable negative reference buffer */
+		ADS114S0X_REGISTER_REF_NOT_REFN_BUF_SET(reference_control, 0b1);
+		/* disable positive reference buffer */
+		ADS114S0X_REGISTER_REF_NOT_REFP_BUF_SET(reference_control, 0b1);
+		/* use internal reference */
+		ADS114S0X_REGISTER_REF_REFSEL_SET(reference_control, 0b10);
+		break;
+	case ADC_REF_EXTERNAL0:
+		/* enable negative reference buffer */
+		ADS114S0X_REGISTER_REF_NOT_REFN_BUF_SET(reference_control, 0b0);
+		/* enable positive reference buffer */
+		ADS114S0X_REGISTER_REF_NOT_REFP_BUF_SET(reference_control, 0b0);
+		/* use external reference 0*/
+		ADS114S0X_REGISTER_REF_REFSEL_SET(reference_control, 0b00);
+		break;
+	case ADC_REF_EXTERNAL1:
+		/* enable negative reference buffer */
+		ADS114S0X_REGISTER_REF_NOT_REFN_BUF_SET(reference_control, 0b0);
+		/* enable positive reference buffer */
+		ADS114S0X_REGISTER_REF_NOT_REFP_BUF_SET(reference_control, 0b0);
+		/* use external reference 0*/
+		ADS114S0X_REGISTER_REF_REFSEL_SET(reference_control, 0b01);
+		break;
+	default:
+		LOG_ERR("reference %i is not supported", channel_cfg->reference);
+		return -EINVAL;
+	}
+
+	if (channel_cfg->differential) {
+		if (channel_cfg->input_positive >= ADS114S0X_INPUT_SELECTION_AINCOM) {
+			LOG_ERR("positive channel input %i is invalid",
+				channel_cfg->input_positive);
+			return -EINVAL;
+		}
+
+		if (channel_cfg->input_negative >= ADS114S0X_INPUT_SELECTION_AINCOM) {
+			LOG_ERR("negative channel input %i is invalid",
+				channel_cfg->input_positive);
+			return -EINVAL;
+		}
+
+		if (channel_cfg->input_positive == channel_cfg->input_negative) {
+			LOG_ERR("negative and positive channel inputs must be different");
+			return -EINVAL;
+		}
+
+		ADS114S0X_REGISTER_INPMUX_MUXP_SET(input_mux, channel_cfg->input_positive);
+		ADS114S0X_REGISTER_INPMUX_MUXN_SET(input_mux, channel_cfg->input_negative);
+	} else {
+		if (channel_cfg->channel_id >= ADS114S0X_INPUT_SELECTION_AINCOM) {
+			LOG_ERR("channel input %i is invalid", channel_cfg->input_positive);
```

Comment:
See:
https://github.com/zephyrproject-rtos/zephyr/blob/4a218aa1990dc7cdd4785d277a52f546cb4013e4/drivers/adc/adc_ads114s0x.c#L511-L513
```suggestion
		if (channel_cfg->input_positive >= ADS114S0X_INPUT_SELECTION_AINCOM) {
			LOG_ERR("channel input %i is invalid", channel_cfg->input_positive);
```

Reply:
done